src/mpid/ch4/netmod/ucx/ucx_pre.h

```diff
@@ -24,9 +24,14 @@ typedef struct {
     ucp_datatype_t ucp_datatype;
 } MPIDI_UCX_dt_t;
 
-typedef union {
+typedef struct {
     ucp_tag_message_h message_handler;
     MPIDI_UCX_ucp_request_t *ucp_request;
+    void *tmp_buf;
+    void *user_buf;
+    MPI_Aint user_count;
+    MPI_Datatype datatype;
+    ucp_dt_iov_t *iov;
 } MPIDI_UCX_request_t;
```

Comment:
I prefer (nested) unions to keep their intent explicit. E.g.
```
typedef union {
    ucp_tag_message_h message_handler;
    struct {
        MPIDI_UCX_ucp_request_t *ucp_request;
        int type;
        union {
            struct {
                void *tmp_buf;
                void *user_buf;
                void *count;
                void *datatype;
            } mpich_unpack;
            struct {
                ucp_dt_iov_t *iov;
            } ucx_iov;
        } u;
    } s;
} MPIDI_UCX_request_t;
```

Reply:
Done.